src/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs

```diff
@@ -151,5 +153,52 @@ public override int GetMatsOsPlatformCode()
             return MatsConverter.AsInt(OsPlatform.Win32);
         }
         protected override IFeatureFlags CreateFeatureFlags() => new NetCoreFeatureFlags();
+
+        public override Task StartDefaultOsBrowserAsync(string url)
```

Comment:
design question.  do we want this to be pluggable / configurable from the outside?  For example, I could see us providing an implementation here that ensures we launch Edge/EdgeChromium in cases where we want PRT access even if it's not the default system browser.  Or to let app developers choose which browser they want (e.g. on linux or macos).  Might be an interesting/useful extensibility point.

Reply:
agree: I'm assuming that would be step 2 of the systembrowser integration, to bake functionality into the library to pick a specific browser?

---
In reply to: [283505817](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1142#discussion_r283505817) [](ancestors = 283505817)

Reply:
Yeah, it's on my todo list. I was thinking to expose a string "format" so that we can users can pass in things like `start microsoft-edge:{0}` where {0} will be the url.